plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/base/impl/FeatureAttribute.java

```diff
@@ -111,27 +107,27 @@ public boolean checkValue() {
 	 * @return true if match, else false.
 	 */
 	public boolean checkValue(String value) {
-		if (type.toString().equals(LONG)) {
-			try {
-				Long.parseLong(value);
+		if (!value.isEmpty()) {
+			if (type.toString().equals(LONG)) {
+				try {
+					Long.parseLong(value);
+				} catch (final NumberFormatException e) {
+					return false;
+				}
 				return true;
-			} catch (final NumberFormatException e) {
-				return false;
 			}
-		}
-		if (type.toString().equals(DOUBLE)) {
-			try {
-				Double.parseDouble(value);
+			if (type.toString().equals(DOUBLE)) {
+				try {
+					Double.parseDouble(value);
+				} catch (final NumberFormatException e) {
+					return false;
+				}
 				return true;
-			} catch (final NumberFormatException e) {
-				return false;
 			}
-		}
-		if (type.toString().equals(BOOLEAN)) {
-			if (value.toLowerCase().equals("true") || value.toLowerCase().equals("false")) {
-				return true;
-			} else {
-				return false;
+			if (type.toString().equals(BOOLEAN)) {
+				if (value.toLowerCase().equals("true") || value.toLowerCase().equals("false")) {} else {
+					return false;
+				}
```

Comment:
This seems to be an incorrect use-case of an else branch. Please use negation instead.